src/AvtaleSide/steg/BeregningTilskudd/visningTilskuddsperioder/VisningTilskuddsperioderTabell.tsx

```diff
@@ -24,55 +24,68 @@ const VisningTilskuddsperioderTabell: React.FC<Properties> = ({ className }: Pro
+                            const nyProsent: boolean =
+                                index > 0
+                                    ? avtale.tilskuddPeriode[index - 1].lonnstilskuddProsent !==
+                                      periode.lonnstilskuddProsent
+                                    : false;
+                            if (index < startIndexVisning || index > sluttIndexVisning) {
+                                return null;
+                            } else if (index !== 0 && (index === startIndexVisning || index === sluttIndexVisning)) {
+                                return (
+                                    <Table.Row key={index}>
+                                        <Table.DataCell textSize={'small'} colSpan={100}>
+                                            ...
+                                        </Table.DataCell>
+                                    </Table.Row>
+                                );
+                            }
+                            return (
+                                <Table.Row
+                                    key={index}
+                                    style={{
+                                        borderTop: nyProsent ? '2px solid gray' : 'undefined',
+                                    }}
+                                >
+                                    <Table.DataCell textSize={'small'}>
+                                        {formaterPeriode(periode.startDato, periode.sluttDato)}
+                                    </Table.DataCell>
+                                    {innloggetBruker.erNavAnsatt && (
+                                        <Table.DataCell textSize={'small'}>
+                                            <EtikettStatus tilskuddsperiodestatus={periode.status} size="small" />
+                                        </Table.DataCell>
+                                    )}
+                                    <Table.DataCell align={'right'} textSize={'small'}>
+                                        {avtale.tiltakstype === 'VARIG_LONNSTILSKUDD' &&
+                                            periode.status !== 'BEHANDLET_I_ARENA' && (
+                                                <>{periode.lonnstilskuddProsent}%</>
+                                            )}
+                                        {(avtale.tiltakstype === 'MIDLERTIDIG_LONNSTILSKUDD' ||
+                                            avtale.tiltakstype === 'SOMMERJOBB') && (
+                                            <>{periode.lonnstilskuddProsent}%</>
+                                        )}
+                                    </Table.DataCell>
+                                    <Table.DataCell textSize={'small'}>{formaterPenger(periode.beløp)}</Table.DataCell>
+                                </Table.Row>
+                            );
+                        })}
+                </Table.Body>
+            </Table>
+
```

Comment:
Logikken her kunne kanskje ha blitt forenklet litt i samme sleng. For eksempel:

```
const tilskuddsperioder = useMemo(() =>
        avtale.tilskuddPeriode
            .filter((p: TilskuddsPeriode) => p.aktiv)
            .slice(startIndexVisning, sluttIndexVisning),
    [avtale.tilskuddPeriode, startIndexVisning, sluttIndexVisning]
);

...

return (
    ...
    {tilskuddsperioder.map((periode: TilskuddsPeriode, index: number) => {
        // Render rader
    })}
    {!visAllePerioder && <div className={cls.element('tabell-innslag')}>...</div>}
    ...
);
```

Da kunne kanskje all if-logikken blitt droppet i map'en? 🤔

Reply:
Det er jo det som er gjort i VTAO tabellen og ser jeg nå 😅